### ISSUE TITLE
Wrong bucket url after uploading video files from Strapi by using `@strapi/provider-upload-aws-s3`

### ISSUE THREAD

**Issue by rb090** [strapi/strapi#21807]
## Bug report

The urls for uploaded video files are wrong saved into the DB.

I use Strapi `v4.25.13` with the plugin `@strapi/provider-upload-aws-s3` in version `4.25.13` to upload any kind of assets to a bucket. 

I am not using AWS buckets. I am using OTC OBS buckets. But they work 100% the same like AWS buckets. We develop several services which were connected to OTC buckets and we were able to use the AWS SDK. 

### Required System information

- Node.js version: 20.4.0
- NPM version: 9.7.2
- Strapi version: 4.25.13
- Database: MySQL
- Operating system: darwin-arm64
- Is your project Javascript or Typescript: Javascript

Running `yarn strapi report` gave me the following output:

```
roxana@MacBook strapi-cms % yarn strapi report
yarn run v1.22.22
warning ../package.json: No license field
$ strapi report
Launched In: 288 ms
Environment: development
OS: darwin-arm64
Strapi Version: 4.25.13
Node/Yarn Version: yarn/1.22.22 npm/? node/v20.4.0 darwin arm64
Edition: Community
Database: mysql
✨  Done in 1.06s.
```
 
### Describe the bug

The uploads are successful for different kind of files (images, pdfs) but for videos, the upload url is wrong. 

For any kind of videos, the files are correctly uploaded to the bucket, but the url in the DB looks sth like this: `https:///<BUCKET-NAME>/myvideo.mp4`. 

Instead of the custom `endpoint` url with the filename I set up in my `plugins.js`.

Is there a chance that my `plugins.js` is somehow wrong and I only need to correct sth?

### Steps to reproduce the behavior

1. Have a Strapi with setup `@strapi/provider-upload-aws-s3` plugin and connected bucket
2. Go to your Strapi instance 
3. Go to media library
4. Upload any kind of video
5. Check `files` table in the database and check the `url` property. It has corrupted url.

### Expected behavior

I expect that the `file.url` corresponds defined `endpoint` in `plugin.js` and the filename is appended at the end -> `https://obs.${env("AWS_REGION")}.otc.t-systems.com/filename.extension`. Like it behaves when uploading images or any other type of files.

### Screenshots

 NONE

### Code snippets

My `plugins.js`:

```
module.exports = ({ env }) => ({
  upload: {
    config: {
      provider: "aws-s3",
      providerOptions: {
        s3Options: {
          credentials: {
            accessKeyId: env("AWS_ACCESS_KEY_ID"),
            secretAccessKey: env("AWS_ACCESS_SECRET"),
          },
          region: env("AWS_REGION"),
          endpoint: `https://obs.${env("AWS_REGION")}.otc.t-systems.com`,
          params: {
            ACL: "private",
            signedUrlExpires: env("AWS_SIGNED_URL_EXPIRES") || 15 * 60,
            Bucket: env("AWS_BUCKET"),
          },
        },
      },
      actionOptions: {
        upload: {},
        uploadStream: {},
        delete: {},
      },
    },
  },
});
```

My `middlewares.js`:

```
module.exports = [
  "strapi::errors",
  {
    name: "strapi::security",
    config: {
      contentSecurityPolicy: {
        useDefaults: true,
        directives: {
          "connect-src": ["'self'", "https:"],
          "img-src": [
            "'self'",
            "data:",
            "blob:",
            `${process.env.AWS_BUCKET}.obs.${process.env.AWS_REGION}.otc.t-systems.com`,
          ],
          "media-src": [
            "'self'",
            "data:",
            "blob:",
            `${process.env.AWS_BUCKET}.obs.${process.env.AWS_REGION}.otc.t-systems.com`,
          ],
          upgradeInsecureRequests: null,
        },
      },
    },
  },
  "strapi::poweredBy",
  "strapi::cors",
  "strapi::logger",
  "strapi::query",
  "strapi::body",
  "strapi::session",
  "strapi::favicon",
  "strapi::public",
];
```

### Additional context

I installed the plugin to my Strapi project with `yarn add @strapi/provider-upload-aws-s3@4.25.13`

**Comment by bolg55** [comment]
Can you just confirm you are using `plugins.js` and not `plugin.js` as your issue suggests?

**Comment by rb090** [comment]
Oh no I am sorry for this. That was a typo (auto correction on Mac I overseen 😞). Yes I confirm that the name of the file is `plugins.js`. I also corrected the issue description regarding this.

**Comment by rb090** [comment]
I also updated the ticket description, I miss to mention that the files are uploaded to the bucket. But the url written in the database in the `files` table, column `url` is wrong.

**Comment by jotapepinheiro** [comment]
```js
export default () => ({
  upload: {
    config: {
      provider: 'aws-s3',
      providerOptions: {
        baseUrl: `https://${process.env.AWS_BUCKET}`,
        rootPath: process.env.AWS_ROOT_PATH || 'uploads',
        s3Options: {
          credentials: {
            accessKeyId: process.env.AWS_ACCESS_KEY_ID,
            secretAccessKey: process.env.AWS_ACCESS_SECRET,
          },
          region: process.env.AWS_REGION,
          params: {
            ACL: process.env.AWS_ACL || 'public-read',
            signedUrlExpires: process.env.AWS_SIGNED_URL_EXPIRES || 15 * 60,
            Bucket: process.env.AWS_BUCKET
          },
        },
        
      },
      actionOptions: {
        upload: {
          autoOrientation: true,
          responsive: [
            {
              name: 'large',
              resize: '1920x1440>',
              quality: 90,
            },
            {
              name: 'medium',
              resize: '960x720>',
              quality: 90,
            },
            {
              name: 'small',
              resize: '480x360>',
              quality: 90,
            },
            {
              name: 'thumbnail',
              resize: '200x200>',
              quality: 70,
            },
          ],
        },
        uploadStream: {},
        delete: {},
      },
    },
  }
});
```

```js
export default [
  'strapi::logger',
  'strapi::errors',
  {
    name: 'strapi::security',
    config: {
      contentSecurityPolicy: {
        useDefaults: true,
        directives: {
          'connect-src': ["'self'", 'https:'],
          'img-src': [
            "'self'",
            'data:',
            'blob:',
            'market-assets.strapi.io',
            `${process.env.AWS_BUCKET}`,
            `s3.${process.env.AWS_REGION}.amazonaws.com`
          ],
          'media-src': [
            "'self'",
            'data:',
            'blob:',
            'market-assets.strapi.io',
            `${process.env.AWS_BUCKET}`,
            `s3.${process.env.AWS_REGION}.amazonaws.com`
          ],
          upgradeInsecureRequests: null,
        },
      },
    },
  },
  'strapi::cors',
  'strapi::poweredBy',
  'strapi::query',
  'strapi::body',
  'strapi::session',
  'strapi::favicon',
  'strapi::public',
];

```

**Comment by rb090** [comment]
Thanks for sharing your `middlewares.js` and `plugins.js` @jotapepinheiro. 

If I change the property `endpoint` to `baseUrl` in `plugins.js` with my `endpoint` value  `https://obs.${env("AWS_REGION")}.otc.t-systems.com` this does not work (`AWS_REGION` is in my case correct region of OTC bucket). My understanding from the plugin `@strapi/provider-upload-aws-s3` documentation is that I have to use always `endpoint` property as I have a non AWS bucket url. 

Beside this I added `'market-assets.strapi.io'` and `obs.${process.env.AWS_REGION}.otc.t-systems.com` to `directives` in `middlewares.js`.

I also upgraded Strapi to `v4.25.14`.

**Comment by rb090** [comment]
It seems that this is not only affecting video files. A wrong url is written in the DB for all files larger 6MB. I still believe that there is some issue within the plugin itself. I tested to upload large files to OTC buckets with all my services using AWS SDK. There are no issues. Is there any payload max size we need to increase somewhere?

**Comment by derrickmehaffy** [comment]
> It seems that this is not only affecting video files. A wrong url is written in the DB for all files larger 6MB. I still believe that there is some issue within the plugin itself. I tested to upload large files to OTC buckets with all my services using AWS SDK. There are no issues. Is there any payload max size we need to increase somewhere?
> 
> Even if that is very weird behaviour, the files are uploaded to the bucket. We only end up in url in DB `file.url` like this `https:///<BUCKET-NAME>/filename.extension`. If there are any limit issues I would rather expect that there is a strapi error and request is not even processed. So file would in that case also not be uploaded...

I still believe this is something specific with your provider as I've tested this quite a lot recently and this doesn't happen on the official S3 (AWS) implementation.

**Comment by derrickmehaffy** [comment]
Yeah please do, if we need to have some kind of capability code added to the provider it would be good to have it.

**Comment by IJsLauw** [comment]
Following this, since I seem to have the same problem with uploads going to a digital ocean spaces bucket.

**Comment by rb090** [comment]
I’m really sorry to hear that, @IJsLauw 😢. But thank you for letting us know as this is really interesting. I actually reached out to OTC last week about this issue, but unfortunately, I haven’t heard back from them yet. As soon as I get an update from their team, I’ll make sure to update this ticket. It might help you as well.
